### PR TITLE
Support killing workers, soft-delete serverside

### PR DESCRIPTION
**Pull request by davepagurek**
- Adds soft-delete to workers so that you can still see finished jobs
  - Sends message to the worker when a soft-delete happens
- Updates ordering so deleted workers go last
- Updates ordering so finished jobs go last
- Updates job dispatch logic so it only assigns to workers for whom we received a healthcheck for recently, and order by the number of *active* jobs to find a free worker instead of *total* jobs

### CODE REVIEW

File: app/controllers/jobs_controller.rb [hunk 1]
```diff
@@ -74,7 +74,11 @@ def set_job
   def find_free_worker
     current_user
       .workers
-      .joins("left outer join jobs on jobs.worker_id = workers.id")
+      .active
+      .joins(
+          "left outer join jobs on jobs.worker_id = workers.id " +
```
Comment (AndrewMcBurney) [inline app/controllers/jobs_controller.rb]: Do you need to write `"left outer join jobs ..."`, or can't you just write `"left join jobs ..."`?

Also, could you capitalize the SQL keywords, please?

Comment (davepagurek) [reply app/controllers/jobs_controller.rb]: Ah yeah just left join works! And will do

Comment (AndrewMcBurney) [reply app/controllers/jobs_controller.rb]: The reason I mentioned that is I found that out when studying for the DB midterm 😂 😂

Comment (davepagurek) [reply app/controllers/jobs_controller.rb]: Truuue. Also I just added a commit from my phone on the Greyhound back to Waterloo lol

**Pull request merged**